rust/README.md

```diff
@@ -0,0 +1,135 @@
+# Practical Networked Applications in Rust
+
+A training course on practical software construction in Rust.
+
+The final project in this course is a networked, persistent [key-value store]
+with multithreading, asynchronous I/O (via [tokio] and [tower]), a
+high-performance network protocol ([GRPC] via [prost] and [tower-grpc]),
+backed by a simple [LSM-tree] that the student will write themselves.
+
+Subjects covered include:
+
+- Structuring and maintaining Rust programs
+- Fun and foolproof parallel programming
+- Asyncronous programming with [futures] and [tokio]
+- High-performance networking with [GRPC] and [prost]
+- Benchmarking with [criterion] and [critcmp]
+- Robust and reliable error handling with [failure]
+- Serialization with [serde]
+- Build scripts and the interaction between build-time and run-time
+- Insights into inner workings of the language and libraries
+- How to learn what you don't know about Rust and find what you need to find
+- ... and etc.
+
+Goals of this course are to convey knowledge relevant to real-world Rust
+programming, including inherited wisdom and best practices; to look "under the
+hood" and understand how and why Rust works like it does; to tour the world of
+Rust documentation and code such that the student can learn how to find answers
+to their questions on their own.
+
+Non-goals of this course include teaching installation, syntax and other Rust
+basics; teaching basic data structures and algorithms; being a comprehensive
+resource on the Rust language. That information is easily found [elsewhere].
+
+The course can be followed and completed online or delivered offline as a series
+of training sessions, and consists of
+
+- _readings_ - background material that is expected to be read before each
```

Comment:
I like this, encouraging people to dig into topics as they learn them is teaching good habits.